js/app/dashboard/components/Tour.js

```diff
@@ -1,45 +1,184 @@
+                          : <Popconfirm
+                              placement="left"
+                              title={ t('ADMIN_DASHBOARD_DELETE_TOUR_CONFIRM') }
+                              onConfirm={ onConfirmDelete }
+                              okText={ t('CROPPIE_CONFIRM') }
+                              cancelText={ t('ADMIN_DASHBOARD_CANCEL') }
+                              >
+                              <a role="button" href="#" className="text-reset"
+                                onClick={ e => e.preventDefault() }>
+                                <i className="fa fa-trash"></i>
+                              </a>
+                          </Popconfirm>
+                     }            
+                </div>
+            </>
+            }
+            { 
+              toggleInputForName && renderEditNameForm() 
+            }
         </h4>
       </div>
       <div id={ `${collapseId}` } className="panel-collapse collapse" role="tabpanel">
-        <ul className="list-group list-group-padded">
-          { tasks.map(task => {
-            return (
-              <Task
-                key={ task['@id'] }
-                task={ task }
-              />
-            )
-          })}
-        </ul>
+        <Droppable 
+            isDropDisabled={!isDroppable}
+            droppableId={ `tour:${tour['@id']}` }
+          >
+            {(provided, snapshot) => (
+              <div
+              className={ classNames({
+                'taskList__tasks': true,
+                'list-group': true,
+                'm-0': true,
+                'p-0': true,
+                'nomargin': true,
+                'taskList__tasks--empty': !tour.items.length
+              }) }
+              style={getDroppableListStyle(snapshot.isDraggingOver)}
+              ref={ provided.innerRef } { ...provided.droppableProps }>
+                { _.map(tour.items, (task, index) => {
+                  return (
+                    <Draggable key={ `task-${task.id}` } draggableId={ `${task['@id']}` } index={ index }>
+                      {(provided) => (
+                        <div
+                          ref={ provided.innerRef }
+                          { ...provided.draggableProps }
+                          { ...provided.dragHandleProps }
+                        >
+                          <Task
+                            key={ task['@id'] }
+                            task={ task }
+                            onRemove={ (taskToRemove) => removeTaskFromTour(tour, taskToRemove, tourIsAssignedTo(tour)) }
+                            />
+                        </div>
+                      )}
+                    </Draggable>
+                  )
+                })}
+                { provided.placeholder }
+              </div>
+            )}
+          </Droppable>
       </div>
     </div>
   )
 }
 
-export default withTranslation()(Tour)
+function mapStateToProps (state) {
```

Comment:
That could be replaced with `useSelector` and `useDispatch` hooks in the function components. https://react-redux.js.org/api/hooks

Reply:
OK I ll look into that, I am not very familiar with hooks yet but we need now to change everywhere for hooks I guess

Reply:
Ideally, yes. But the same as the previous one, not sure if we get time for this. For now, I would at least avoid using it in the new code